pipeline-adapters/mlrun-pipelines-kfp-common/src/mlrun_pipelines/common/client.py

```diff
@@ -0,0 +1,186 @@
+# limitations under the License.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+
+class AbstractClient(ABC):
+    """
+    Abstract Base Class for the Kubeflow Pipelines Client.
+
+    This class defines the public interface for interacting with Kubeflow Pipelines.
+    """
+
+    @abstractmethod
+    def get_kfp_healthz(self):
+        """
+        Retrieve the healthz status of the KFP API.
+
+        This method retries multiple times until the endpoint responds or a timeout occurs.
+
+        :return: A valid ApiGetHealthzResponse if successful, otherwise None.
+        :raises TimeoutError: If the endpoint is not reachable after the specified retries.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def create_experiment(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        namespace: Optional[str] = None,
+    ):
+        """
+        Create a new experiment if it does not already exist.
+
+        This method searches for an experiment by name (and optional namespace), and creates
+        it if not found. If found, the existing experiment is returned.
+
+        :param name:         The name of the experiment to create or retrieve.
+        :param description:  A description for the experiment.
+        :param namespace:    An optional Kubernetes namespace.
+        :return: An ApiExperiment object representing the experiment.
+        :raises ValueError:  If multiple experiments with the same name are found.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_experiment(
+        self,
+        experiment_id: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        namespace: Optional[str] = None,
+    ):
+        """
+        Retrieve an experiment by ID or name.
+
+        This method fetches an experiment using its unique ID or by searching for the name
+        (optionally in a specified namespace).
+
+        :param experiment_id:   The ID of the experiment to retrieve.
+        :param experiment_name: The name of the experiment to retrieve.
+        :param namespace:       An optional Kubernetes namespace for filtering by name.
+        :return: An ApiExperiment object representing the experiment.
+        :raises ValueError: If neither experiment_id nor experiment_name is provided, or if
+                            multiple experiments share the same name.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def run_pipeline(
+        self,
+        experiment_id: str,
+        job_name: str,
+        pipeline_package_path: Optional[str] = None,
+        params: Optional[dict[str, Any]] = None,
+        pipeline_id: Optional[str] = None,
+        version_id: Optional[str] = None,
+        pipeline_root: Optional[str] = None,
+        enable_caching: Optional[bool] = None,
+        service_account: Optional[str] = None,
+    ):
```

Comment:
```suggestion
    ) -> kfp_server_api.ApiRun :
```
Add in all functions

Reply:
`kfp_server_api.ApiRun` only exists in KFP 1.8, and this abstract class will also serve the KFP 2.4 client, so I can't use this return type annotation.